packages/docs/src/routes/docs/components/context/index.mdx

```diff
@@ -72,34 +97,43 @@ Let's dig into every API involved:
 
 ### `createContextId()`
 
-This method takes a string that gives the context unique name, if your application uses a lot of different context or you are writing a component library, we recommend to follow a name convention that avoids conflicts, such as:
+This method is used to create a new `Context`.
 
 ```tsx
-export const QwikCityContext = createContextId('io.builder.qwik.city');
+export interface GenericType {
+  ...
+}
+
+export const QwikCityContext = createContextId<GenericType>(name: string): ContextId<GenericType>;
 ```
 
-Notice that the value returned by `createContextId()` does not hold any state, and it's immutable. It's only used to describe the name and type of the context, like an address or an identifier.
+#### Parameters
 
-Because it does not hold any state, it's ok to call it and make it a singleton, exported in some shared module.
+- `name`: it is a unique string given to `createContextId` as an identifier of context. This will avoid conflicts when there are multiple contexts. It is advised to use a naming convention like `io.builder.qwik.city`.
 
-### `useContextProvider()`
+#### Returns
+
+Notice that the value returned by `createContextId()` does not hold any state, it is a immutable object i.e. `{ id: 'io.builder.qwik.city' }`. It's only used to describe the name and type of the context, like an address or an identifier.
 
-Like all use- methods, it can only be called at the root of [`component$()`](/docs/components/overview/index.mdx#component) (not inside branches). This method is called by some higher level component and it's what assigns (provides) a value to the context. The provided value will not be globally available across the whole render tree, but only to descendant components in the tree.
+Since, it does not hold any state, it's ok to call it and make it a singleton, exported in some shared module.
 
-The value passed to [`useContextProvider()`](/docs/components/context/index.mdx#usecontextprovider) can be any primitive, object (including from useStore), or array that contains serializable values.
+### `useContextProvider()`
+
+- This method is used to provide a value to `Context` after using [`createContextId()`](#createContextId).
```

Comment:
I'd change it to :

This method is used to create a `Context` for a specific component and its descendants, using the `ContextId` as the key identifier of the context.